packages/flutter/lib/src/rendering/object.dart

```diff
@@ -819,6 +834,146 @@ class PipelineOwner {
     profile(() { Timeline.finishSync(); });
   }
 
+  /// The maximum number of iterations to use when checking for potential
+  /// incorrect uses of elevation on a [RenderPhysicalModel] or
+  /// [RenderPhysicalShape] with regard to painting order. This number is not
+  /// the same as the number of total [RenderObject]s in the tree, but rather the
+  /// total number of [RenderPhysicalModel] or [RenderPhysicalShape]s to check.
+  ///
+  /// This property is only used when asserts are enabled (e.g. debug mode).
+  ///
+  /// The default value is 10.
+  ///
+  /// See also [debugCheckElevationData].
+  int get maxElevationObjectsToCheck => _maxElevationObjectsToCheck;
+  int _maxElevationObjectsToCheck = 10;
+
+  /// Sets the maximum number of iterations to use when checking for potential
+  /// incorrect uses of elevation on a [RenderPhysicalModel] or
+  /// [RenderPhysicalShape] with regard to painting order. This number is not
+  /// the same as the number of total [RenderObject]s in the tree, but rather the
+  /// total number of [RenderPhysicalModel] or [RenderPhysicalShape]s to check.
+  ///
+  /// Setting this number too high will degrade performance if many render
+  /// objects using elevation are present in the tree all are used correctly.
+  /// See [debugCheckElevationData] for a discussion of the performance
+  /// characterstics of the check.
+  ///
+  /// This property is only used when asserts are enabled (e.g. debug mode).
+  ///
+  /// The default value is 10.
+  ///
+  /// This value must be >= 0 and must not be null. Setting it to 0 will
+  /// disable the check, and a warning will be printed once if there are more
+  /// objects available than actually being checked.
+  set maxElevationObjectsToCheck(int value) {
+    assert(value != null);
+    assert(value >= 0);
+    _maxElevationObjectsToCheck = value;
+  }
+
+  Map<RenderObject, List<_ElevationData>> _elevations;
+  Path _translatePath(Path p, Offset offset) {
+    final Matrix4 matrix = Matrix4.identity()..translate(offset.dx, offset.dy);
```

Comment:
his probably also need to take other paint transformations into consideration? (`applyPaintTransform` maybe)

Reply:
Done.

I could use some feedback on whether I'm right to ignore `RenderView`.  That one ends up applying th edevicepixelratio to the transform, and it messes things up.  But I'm not sure how else it might get used.